ChatQuickstart/QuickstartChatManager.swift

```diff
@@ -0,0 +1,141 @@
+//
+//  QuickstartChatManager.swift
+//  ChatQuickstart
+//
+//  Created by Jeffrey Linwood on 3/11/20.
+//  Copyright © 2020 Twilio, Inc. All rights reserved.
+//
+
+import UIKit
+
+import TwilioChatClient
+
+class QuickstartChatManager:NSObject, TwilioChatClientDelegate {
+    
+    
+    // Important - update this URL with your Twilio Function URL
+    let tokenURL = "https://YOUR_TWILIO_FUNCTION_DOMAIN_HERE.twil.io/chat-token"
+    
+    var chatManagerDelegate: QuickstartChatManagerDelegate?
+    
+    // MARK: Chat variables
+    var client: TwilioChatClient? = nil
+    var generalChannel: TCHChannel? = nil
+    var messages: [TCHMessage] = []
+    
+    func chatClient(_ client: TwilioChatClient, synchronizationStatusUpdated status: TCHClientSynchronizationStatus) {
+        if status == .completed {
+            // Join (or create) the general channel
+            let defaultChannel = "general"
+            if let channelsList = client.channelsList() {
```

Comment:
there's `createGeneralChannelIfNeeded` function below that perhaps could be used here?

Reply:
I refactored this entire quick start manager to reduce the nesting, and to separate the functionality. Can you take another look at it? Thanks!